internal/store/horizontalpodautoscaler.go

```diff
@@ -134,51 +134,53 @@ func hpaMetricFamilies(allowAnnotationsList, allowLabelsList []string) []generat
 				for _, m := range a.Spec.Metrics {
 					var metricName string
 
-					var v [metricTargetTypeCount]int64
+					var v [metricTargetTypeCount]float64
 					var ok [metricTargetTypeCount]bool
 
 					switch m.Type {
 					case autoscaling.ObjectMetricSourceType:
 						metricName = m.Object.Metric.Name
 
-						v[value], ok[value] = m.Object.Target.Value.AsInt64()
+						if m.Object.Target.Value != nil {
+							v[value], ok[value] = float64(m.Object.Target.Value.MilliValue())/1000, true
+						}
 						if m.Object.Target.AverageValue != nil {
-							v[average], ok[average] = m.Object.Target.AverageValue.AsInt64()
+							v[average], ok[average] = float64(m.Object.Target.AverageValue.MilliValue())/1000, true
 						}
 					case autoscaling.PodsMetricSourceType:
 						metricName = m.Pods.Metric.Name
 
-						v[average], ok[average] = m.Pods.Target.AverageValue.AsInt64()
+						v[average], ok[average] = float64(m.Pods.Target.AverageValue.MilliValue())/1000, true
 					case autoscaling.ResourceMetricSourceType:
 						metricName = string(m.Resource.Name)
 
-						if ok[utilization] = (m.Resource.Target.AverageUtilization != nil); ok[utilization] {
-							v[utilization] = int64(*m.Resource.Target.AverageUtilization)
+						if m.Resource.Target.AverageUtilization != nil {
+							v[utilization], ok[utilization] = float64(*m.Resource.Target.AverageUtilization), true
 						}
 
 						if m.Resource.Target.AverageValue != nil {
-							v[average], ok[average] = m.Resource.Target.AverageValue.AsInt64()
+							v[average], ok[average] = float64(m.Resource.Target.AverageValue.MilliValue())/1000, true
 						}
 					case autoscaling.ExternalMetricSourceType:
 						metricName = m.External.Metric.Name
 
 						if m.External.Target.Value != nil {
-							v[value], ok[value] = m.External.Target.Value.AsInt64()
+							v[value], ok[value] = float64(m.External.Target.Value.MilliValue())/1000, true
 						}
 						if m.External.Target.AverageValue != nil {
-							v[average], ok[average] = m.External.Target.AverageValue.AsInt64()
+							v[average], ok[average] = float64(m.External.Target.AverageValue.MilliValue())/1000, true
 						}
 					default:
 						// Skip unsupported metric type
 						continue
 					}
 
-					for i := range ok {
+					for i := range v {
```

Comment:
I assume modifying the slice does not change the behavior because the length of both `v` and `ok` is the same. Is that correct?

Reply:
Yes. Because I removed `ok` during the coding process and forgot to change it back. 

I think it's better to use `ok` here. It has been changed back now.